pkg/stanza/operator/parser/container/parser.go

```diff
@@ -278,7 +279,11 @@ func (p *Parser) extractk8sMetaFromFilePath(e *entry.Entry) error {
 		return fmt.Errorf("type '%T' cannot be parsed as log path field", logPath)
 	}
 
-	parsedValues, err := helper.MatchValues(rawLogPath, pathMatcher)
+	if p.filepathPattern == nil {
```

Comment:
Can we set this once when creating the default configuration at `NewConfigWithID`?

Reply:
Same answer as the previous comment; the patterns are defined in `parser.go`, so if we want to set the pattern at the parser creation, we need to refactor a bit.